control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -761,11 +763,1161 @@ func TestProxyConfigurationDelete(t *testing.T) {
 	}
 }
 
-// TODO
-// func TestUpstreamsWrite(t *testing.T)
+// TestUpstreamsWrite does a subsampling of tests covered in TestProcessUpstreams to make sure things are hooked up
+// correctly. For the sake of test speed, more exhaustive testing is performed in TestProcessUpstreams.
+func TestUpstreamsWrite(t *testing.T) {
+	t.Parallel()
 
-// TODO
-// func TestUpstreamsDelete(t *testing.T)
+	const podName = "pod1"
+
+	cases := []struct {
+		name                    string
+		pod                     func() *corev1.Pod
+		expected                *pbmesh.Upstreams
+		expErr                  string
+		configEntry             func() api.ConfigEntry
+		consulUnavailable       bool
```

Comment:
Same question here. I didn't see anywhere in the test that consumes this.

Reply:
nope, not used